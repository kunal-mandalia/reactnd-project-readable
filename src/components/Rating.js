import React from 'react'
import PropTypes from 'prop-types'
import ChevronUp from 'react-icons/lib/fa/chevron-up'
import ChevronDown from 'react-icons/lib/fa/chevron-down'
import '../styles/Rating.css'

const Rating = ({ rating, onUpvote, onDownvote }) => (
  <div className='rating'>
    <div id='upvote-button' onClick={onUpvote}>
      <ChevronUp size={20} color='grey' className='vote' />
    </div>
    <div id='rating-value' className='rating-value'>
      {rating}
    </div>
    <div id='downvote-button' onClick={onDownvote}>
      <ChevronDown size={20} color='grey' className='vote' />
    </div>
  </div>
)

Rating.propTypes = {
  rating: PropTypes.number.isRequired,
  onUpvote: PropTypes.func.isRequired,
  onDownvote: PropTypes.func.isRequired
}

export default Rating
