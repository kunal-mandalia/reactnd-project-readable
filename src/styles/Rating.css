.rating {
  text-align: center;
  align-self: center;
}

.rating-value {
  font-size: larger;
}

.vote:hover {
  color: black !important;
  cursor: pointer;
}
