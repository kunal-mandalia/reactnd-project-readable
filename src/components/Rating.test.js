import React from 'react'
import { shallow } from 'enzyme'
import Rating from './Rating'
import renderer from 'react-test-renderer'

const mockFnUpvote = jest.fn()
const mockFnDownvote = jest.fn()

const props = {
  rating: 4,
  onUpvote: mockFnUpvote,
  onDownvote: mockFnDownvote,
}

describe(`<Rating .../>`, () => {
  it(`should render rating`, () => {
    const wrapper = shallow(<Rating {...props} />)
    expect(wrapper.find('#rating-value').text()).toContain(props.rating)
  })

  it(`should call increment vote on upvote`, () => {
    const wrapper = shallow(<Rating {...props} />)
    wrapper.find('#upvote-button').simulate('click')
    expect(mockFnUpvote.mock.calls).toHaveLength(1)
    mockFnUpvote.mockClear()
  })

  it(`should call decrement vote on downvote`, () => {
    const wrapper = shallow(<Rating {...props} />)
    wrapper.find('#downvote-button').simulate('click')
    expect(mockFnDownvote.mock.calls).toHaveLength(1)
    mockFnDownvote.mockClear()
  })

  it(`should match snapshot`, () => {
    const tree = renderer.create(<Rating {...props} />)
    expect(tree).toMatchSnapshot()
  })
})
