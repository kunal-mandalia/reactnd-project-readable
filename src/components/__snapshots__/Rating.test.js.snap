// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Rating .../> should match snapshot 1`] = `
<div
  className="rating"
>
  <div
    id="upvote-button"
    onClick={[Function]}
  >
    <svg
      className="vote"
      fill="currentColor"
      height={20}
      preserveAspectRatio="xMidYMid meet"
      style={
        Object {
          "color": "grey",
          "verticalAlign": "middle",
        }
      }
      viewBox="0 0 40 40"
      width={20}
    >
      <g>
        <path
          d="m37.6 29.7l-3.7 3.7q-0.5 0.4-1 0.4t-1-0.4l-11.9-11.9-11.8 11.9q-0.5 0.4-1.1 0.4t-1-0.4l-3.7-3.7q-0.4-0.4-0.4-1t0.4-1l16.6-16.6q0.4-0.4 1-0.4t1 0.4l16.6 16.6q0.4 0.4 0.4 1t-0.4 1z"
        />
      </g>
    </svg>
  </div>
  <div
    className="rating-value"
    id="rating-value"
  >
    4
  </div>
  <div
    id="downvote-button"
    onClick={[Function]}
  >
    <svg
      className="vote"
      fill="currentColor"
      height={20}
      preserveAspectRatio="xMidYMid meet"
      style={
        Object {
          "color": "grey",
          "verticalAlign": "middle",
        }
      }
      viewBox="0 0 40 40"
      width={20}
    >
      <g>
        <path
          d="m37.6 18l-16.6 16.6q-0.4 0.4-1 0.4t-1-0.4l-16.6-16.6q-0.4-0.4-0.4-1t0.4-1l3.7-3.7q0.5-0.4 1-0.4t1 0.4l11.9 11.9 11.9-11.9q0.4-0.4 1-0.4t1 0.4l3.7 3.7q0.4 0.4 0.4 1t-0.4 1z"
        />
      </g>
    </svg>
  </div>
</div>
`;
